### PR TITLE
Add support for using ElastiCache Cluster replica nodes for reads

### PR DESCRIPTION
**Pull request by adeelejaz**
Hi,

We were hitting the bandwidth for the ElastiCache so I implemented a small extension that takes the Primary and replica endpoints and starts sending reads to replica nodes and writes to primary nodes.

We thought it might be useful to someone else so just sharing, it has been working pretty great.

Edit: Just want to reference the orignial source for doing this: https://magento.stackexchange.com/a/122571

### CODE REVIEW

File: Cm/Cache/Backend/Redis.php [hunk 1]
```diff
@@ -360,6 +364,53 @@ protected function _applyClientOptions(Credis_Client $client, $forceSelect = FAL
         }
     }
 
+    protected function _setupReadWriteCluster($options) {
+        $clusterNodes = array();
+
+        if (array_key_exists('master', $options['cluster']) && !empty($options['cluster']['master'])) {
```
Comment (Xon) [inline Cm/Cache/Backend/Redis.php]: fyi, you con't need the array_key_exists test if you have the !empty test

**Comment by colinmollenhour** [comment]
Thanks!
